OpenChange/MAPIStoreMailMessage.m

```diff
@@ -316,15 +316,23 @@ - (void) _fetchHeaderData
           if (i == 0)
              {
                ASSIGN (headerMimeType, mimeType);
-               ASSIGN (headerCharset, charset);               
                parameters = partParameters;
              }
-          else
+
+          if ([mimeType isEqualToString: @"text/plain"])
             {
-              /* We can only put one header charset so we will use utf-8
-                 when we have different charsets in the parts */
-              if (headerCharset != charset)
-                ASSIGN (headerCharset, @"utf-8");
+              //  utf-8 is always encoded in text/plain
+              ASSIGN (headerCharset, @"utf-8");              
```

Comment:
why? If we have text, the charset could be, e.g., latin1. We should set headerCharset as latin1 in that case. I'd say this whole if condition should be removed